### PR TITLE
feat: turbine-core app init

### PR DESCRIPTION
**Pull request by jmar910**
+ Adds an Init function to turbine-core app package for initializing apps from templates
+ Adds templates directory to turbine-core directly
+ Removes templates directory from turbine_rb (currently unused anyway in published gem) and bumps to v0.2.0
+ Adds tests

### CODE REVIEW

**Comment by dianadoherty** [comment]
Also, can you modify the fixture a little? https://github.com/meroxa/turbine-core/blob/e68d75fbd2f2fcfaee7f859aed1239baef14cfb3/lib/ruby/turbine_rb/lib/templates/app/fixtures/demo.json#L2 
`events` should be `collection_name`

File: pkg/app/init.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		if err != nil {
+			return err
+		}
+	}
+
+	// TODO: Maybe write a recursive function that could take care of nested directories like this one
+	err = a.createFixtures()
+	if err != nil {
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (jmar910) [reply pkg/app/init.go:127]: yes we do, wanted to talk with you today on the best way to approach this, lets huddle after standup

File: pkg/app/init.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+)
+
+type AppInit struct {
+	AppName  string
+	Language string
+	Path     string
+}
+
+// AppInitTrait will be used to replace data evaluations provided by the user
+type AppInitTrait struct {
```
[the remark this answers is not in the document]
Comment (jmar910) [reply pkg/app/init.go]: ✅

File: pkg/app/init.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+import (
+	"embed"
+	"html/template"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type AppInit struct {
```
[the remark this answers is not in the document]
Comment (jmar910) [reply pkg/app/init.go:12]: ✅

**Review by dianadoherty — APPROVED** [review]

**Review by dianadoherty — APPROVED** [review]

**Pull request merged**